luna/featurevis/featurevis.py

```diff
@@ -148,33 +67,33 @@ def compute_loss(input_image, model, filter_index, channels_first=False):
     Returns:
         number: the loss for the specified setting
     """
-    activation = model(input_image)
-    # We avoid border artifacts by only involving non-border pixels in the loss.
-    if channels_first:
-        filter_activation = activation[:, filter_index, 2:-2, 2:-2]
+    with rg.gradient_override_map(
+        {'Relu': rg.redirected_relu_grad,'Relu6': rg.redirected_relu6_grad}):
+        activation = model(input_image)
+    if tf.compat.v1.keras.backend.image_data_format() == "channels_first":
+        filter_activation = activation[:, filter_index, :, :]
     else:
-        filter_activation = activation[:, 2:-2, 2:-2, filter_index]
+        filter_activation = activation[:, :, :, filter_index]
     return tf.reduce_mean(filter_activation)
 
 
-@tf.function
-def gradient_ascent_step(img, model, filter_index, learning_rate, channels_first=False):
+@tf.function(experimental_relax_shapes=True)
```

Comment:
What is that? Can you add a comment?

Reply:
Not necessary anymore. I used it to reduce memory usage, memory is not a problem now.